eZ/Publish/Core/Search/Legacy/Tests/Content/HandlerContentTest.php

```diff
@@ -1436,4 +1437,11 @@ public function testFieldRelationFilterInArrayNotMatch()
             )
         );
     }
+
+    public function testGetNonExistingFieldDefinition()
+    {
+        $this->expectException(\eZ\Publish\Core\Base\Exceptions\NotFoundException::class);
+
+        $this->getContentTypeHandler()->getFieldDefinition(0, Type::STATUS_DEFINED);
+    }
```

Comment:
This unit test relies directly on a database. It's deprecated as it belongs to the scope of integration test.
Please try to reproduce the behavior on API level and place it in either `\eZ\Publish\API\Repository\Tests\ContentServiceTest` or \eZ\Publish\API\Repository\Tests\ContentTypeServiceTest` depending on to which API it belongs.

Reply:
Now it is expected that `loadFieldDefinition` will always return an array. Please check https://github.com/ezsystems/ezpublish-kernel/pull/2978/commits/ef12e9ade2507c021afb982654fae6d01a78d4e6.

Regarding moving the test to `\eZ\Publish\API\Repository\Tests\ContentServiceTest` or `\eZ\Publish\API\Repository\Tests\ContentTypeServiceTest` I`m a bit confused. In this test we are testing `getFieldDefinition` method of Content Type handler, which comes from https://github.com/ezsystems/ezpublish-kernel/blob/master/eZ/Publish/Core/Search/Legacy/Tests/Content/AbstractTestCase.php#L78-L93. And that method is not available in ContentService or ContentTypeService. Content type service has the method with the same name, but it does a different thing. Maybe that was a reason why you were suggesting to move this test there? Otherwise, please more detailed suggestion

Reply:
Basically, we are testing ContentTypeHandler here by not ContenTypeServcie. So that's why it seems like this test should be in the current place.

Reply:
Hmm, ok. Did more digging here and it seems it's not reproducible on API level. 
The issue with those tests is that they are executed on in-memory SQLite only, so don't test what they should. But this is a larger issue for tests and minor issue here since it exists only on Persistence handler level, so I'll allow it.